File: components/button/android/CiataButtonExample.kt
package org.ciata.ds.button

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.height
import androidx.compose.material3.MaterialTheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

@Composable
fun CiataButtonExample(
    saving: Boolean,
    onSave: () -> Unit,
    onDelete: () -> Unit,
) {
    MaterialTheme {
        Column {
            CiataButton(
                label = "Salvar alterações",
                onClick = onSave,
                loading = saving,
            )

            Spacer(Modifier.height(16.dp))

            CiataButton(
                label = "Excluir conta",
                variant = CiataButtonVariant.Danger,
                onClick = onDelete,
            )
        }
    }
}

File: components/button/flutter/ciata_button_example.dart
import 'package:flutter/material.dart';

import 'ciata_button.dart';

class CiataButtonExample extends StatefulWidget {
  const CiataButtonExample({super.key});

  @override
  State<CiataButtonExample> createState() => _CiataButtonExampleState();
}

class _CiataButtonExampleState extends State<CiataButtonExample> {
  bool loading = false;

  Future<void> _save() async {
    if (loading) return;

    setState(() => loading = true);
    await Future<void>.delayed(const Duration(seconds: 2));
    if (!mounted) return;
    setState(() => loading = false);
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      children: [
        CiataButton(
          label: 'Salvar alterações',
          loading: loading,
          onPressed: _save,
        ),
        const SizedBox(height: 12),
        CiataButton(
          label: 'Cancelar',
          variant: CiataButtonVariant.secondary,
          onPressed: () {},
        ),
        const SizedBox(height: 12),
        CiataButton(
          label: 'Excluir arquivo',
          variant: CiataButtonVariant.danger,
          onPressed: () {},
        ),
      ],
    );
  }
}
